app/services/daily_xml_to_db_v2.rb

```diff
@@ -0,0 +1,205 @@
+      pp_ok "ht_namespace is #{ht_namespace} and ht_barcode is #{ht_barcode}"
+
+      volume = node1.xpath(".//MODS:detail[@type='volume']/MODS:number/text()").to_s
+      pp_ok "volume value is #{volume}"
+
+      issue_no = node1.xpath(".//MODS:detail[@type='issue']/MODS:number/text()").to_s
+      pp_ok "issue_no value is #{issue_no}"
+
+      edition = node1.xpath(".//MODS:detail[@type='edition']/MODS:number/text()").to_s
+      pp_ok "edition value is #{edition}"
+
+      date_issued = node1.xpath(".//MODS:dateIssued/text()").to_s
+      pp_ok "dateIssued value is #{date_issued}"
+
+      newspaper = node1.xpath("@LABEL").to_s
+      newspaper = newspaper.split(",").first.strip
+      pp_ok "newspaper is #{newspaper}"
+
+      # Check for duplicate dates in array track_dates
+      issue_sequence = 1
+      issue_key_to_check = ht_namespace + key_separater + ht_barcode + key_separater + date_issued + key_separater + issue_sequence.to_s
+      until ((track_issue_keys.bsearch { |id| id == issue_key_to_check}).nil?) do
+        pp "WE HAVE A DUPLICATE: " + issue_key_to_check
+        issue_sequence += 1
+        issue_key_to_check = ht_namespace + key_separater + ht_barcode + key_separater + date_issued + key_separater + issue_sequence.to_s
+      end
+
+      track_issue_keys.push(issue_key_to_check).sort
+
+
+      issue_id = add_data_issue(ht_namespace, ht_barcode, volume, issue_no, edition, date_issued, issue_sequence,  @pub_id)
+
+      pp_ok "ISSUE ID IS: #{issue_id} and issue_sequence IS: #{issue_sequence}"
+
+      # page information -------------
+      # note: page_id is autoincremented by the db
+
+      pages_target = ".//METS:structMap/METS:div[@TYPE='np:issue'][@DMDID='issueModsBib']/METS:div[@TYPE='np:page']"
+
+      pages = @doc.xpath(pages_target, "METS" => "http://www.loc.gov/METS/")
+
+      @doc.xpath(pages_target).each do |node2|
+
+        pp_ok "PAGE INFO ------------------------------------------"
+
+        # pp node2
+        # pp_ok "issue_id value is #{issue_id}"
+
+        page_no = node2.xpath("@ORDERLABEL").to_s
+        pp_ok "page_no value is #{page_no}"
+
+        sequence = node2.xpath("@ORDER").to_s.to_i
+        pp_ok "sequence value is #{sequence}"
+
+        text_link_a = node2.xpath("./METS:mptr[1]")
+        text_link_b = text_link_a.xpath("@xlink:href", "xmlns:xlink"=>"http://www.w3.org/1999/xlink")
+        pp_ok "text_link value is #{text_link_b}"
+
+        img_link_a = node2.xpath("./METS:mptr[2]")
+        img_link_b = img_link_a.xpath("@xlink:href", "xmlns:xlink"=>"http://www.w3.org/1999/xlink")
+        pp_ok "text_link value is #{img_link_b}"
+        
+        add_data_page(issue_id, page_no, sequence, text_link_b, img_link_b)
+
+      end # each page
+    end # each issue
+
+
+  pp "File #{f} processed"
+
+  end # parse_file_to_ar
+
+  def add_daily_publication
+    p = Publication.find_or_create_by(slug: "the-michigan-daily") do |publication|
+      p.title = "The Michigan Daily"
```

Comment:
The variable you have in scope here is `publication` --- the `find_or_create` is passing the object it will create to the parameters of the block you're providing.